dask/blockwise.py

```diff
@@ -241,23 +289,36 @@ def __init__(
         output: str,
         output_indices: Iterable[str],
         dsk: Mapping[str, tuple],
-        indices: Iterable[Tuple[str, Optional[Iterable[str]]]],
+        indices: Iterable[Tuple[Union[str, BlockwiseDep], Optional[Iterable[str]]]],
         numblocks: Mapping[str, Sequence[int]],
         concatenate: bool = None,
         new_axes: Mapping[str, int] = None,
         output_blocks: Set[Tuple[int, ...]] = None,
         annotations: Mapping[str, Any] = None,
-        io_deps: Optional[Mapping[str, Union[dict, tuple]]] = None,
+        io_deps: Optional[Mapping[str, BlockwiseDep]] = None,
     ):
         super().__init__(annotations=annotations)
         self.output = output
         self.output_indices = tuple(output_indices)
         self.output_blocks = output_blocks
         self.dsk = dsk
-        self.indices = tuple(
-            (name, tuple(ind) if ind is not None else ind) for name, ind in indices
-        )
+
+        # Remove `BlockwiseDep` arguments from input indices
+        # and add them to `self.io_deps`.
+        # TODO: Remove `io_deps` and handle indexable objects
+        # in `self.indices` throughout `Blockwise`.
+        self.indices = []
         self.numblocks = numblocks
+        self.io_deps = io_deps or {}
```

Comment:
Does it ever make sense to have some `io_deps` specified via indices, and others via `io_deps`? Or do we view this mostly as a bridge to removing `io_deps`?

Reply:
>Or do we view this mostly as a bridge to removing io_deps?

Right - This is mostly a bridge.  I don't think it makes sense for io_deps to be populated if there are BlockwiseDep objects in `indices` - But I haven't really though carefully about more-complex workflows with fusion, etc.